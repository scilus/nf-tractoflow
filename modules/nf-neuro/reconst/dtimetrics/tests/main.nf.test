nextflow_process {

    name "Test Process RECONST_DTIMETRICS"
    script "../main.nf"
    process "RECONST_DTIMETRICS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "reconst"
    tag "reconst/dtimetrics"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "DWIss1000-dir32.zip", "segmentation.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("reconst - dtimetrics") {
        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        dwi: it.simpleName == "DWIss1000-dir32"
                        segmentation: it.simpleName == "segmentation"
                    }
                ch_dwi = ch_split_test_data.dwi.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec")
                    ]
                }
                ch_mask = ch_split_test_data.segmentation.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/brainmask/brainmask.nii.gz")
                    ]
                }
                input[0] = ch_dwi
                    .join(ch_mask)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    niftiMD5SUM(process.out.ad.get(0).get(1), 6),
                    file(process.out.evecs.get(0).get(1)).name,
                    file(process.out.evecs_v1.get(0).get(1)).name,
                    file(process.out.evecs_v2.get(0).get(1)).name,
                    file(process.out.evecs_v3.get(0).get(1)).name,
                    niftiMD5SUM(process.out.evals.get(0).get(1), 6),
                    niftiMD5SUM(process.out.evals_e1.get(0).get(1), 6),
                    niftiMD5SUM(process.out.evals_e2.get(0).get(1), 6),
                    niftiMD5SUM(process.out.evals_e3.get(0).get(1), 6),
                    niftiMD5SUM(process.out.fa.get(0).get(1), 6),
                    niftiMD5SUM(process.out.ga.get(0).get(1), 6),
                    niftiMD5SUM(process.out.rgb.get(0).get(1), 6),
                    niftiMD5SUM(process.out.md.get(0).get(1), 6),
                    file(process.out.mode.get(0).get(1)).name,
                    niftiMD5SUM(process.out.norm.get(0).get(1), 6),
                    niftiMD5SUM(process.out.rd.get(0).get(1), 6),
                    file(process.out.tensor.get(0).get(1)).name,
                    niftiMD5SUM(process.out.nonphysical.get(0).get(1), 6),
                    niftiMD5SUM(process.out.pulsation_std_dwi.get(0).get(1), 6),
                    niftiMD5SUM(process.out.pulsation_std_b0.get(0).get(1), 6),
                    niftiMD5SUM(process.out.residual.get(0).get(1), 2),
                    process.out.residual_iqr_residuals,
                    process.out.residual_mean_residuals,
                    process.out.residual_q1_residuals,
                    process.out.residual_q3_residuals,
                    process.out.residual_residuals_stats,
                    process.out.residual_std_residuals,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("reconst - dtimetrics_with_b0mask") {
        config "./nextflow_light.config"
        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        dwi: it.simpleName == "DWIss1000-dir32"
                        segmentation: it.simpleName == "segmentation"
                    }
                input[0] = ch_split_test_data.dwi.map{
                    test_data_directory -> [
                        [ id:'test' ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec"),
                        []
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    niftiMD5SUM(process.out.fa.get(0).get(1), 6),
                    process.out.versions
                ).match() }
            )
        }
    }
}
