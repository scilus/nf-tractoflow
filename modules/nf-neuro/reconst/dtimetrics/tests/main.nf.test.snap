{
    "reconst - dtimetrics_with_b0mask": {
        "content": [
            "test__fa.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,cd80bf54bc8529c97cb042d49c50d4fc",
            [
                "versions.yml:md5,609a8b62c7dc862692683ec5860ba677"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-30T19:56:40.272861"
    },
    "reconst - dtimetrics": {
        "content": [
            "test__ad.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,a4f053c9d65bed0062f7778818c555b4",
            "test__evecs.nii.gz",
            "test__evecs_v1.nii.gz",
            "test__evecs_v2.nii.gz",
            "test__evecs_v3.nii.gz",
            "test__evals.nii.gz:md5:header,6a667c1adb8ac1636526963bb9091ea6,data,194ed3036c73a0120615e95bc343607e",
            "test__evals_e1.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,a4f053c9d65bed0062f7778818c555b4",
            "test__evals_e2.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,ffa92eecb19b6ffbfb27d1696e521d44",
            "test__evals_e3.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,7a79a34de17b0653d10242455f3b8a3a",
            "test__fa.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,cd80bf54bc8529c97cb042d49c50d4fc",
            "test__ga.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,a76a594121a0ef669616621eb15aec57",
            "test__rgb.nii.gz:md5:header,952bd2b13b64d7f1b128af7d85191a18,data,a2063164b39833b2a1596683b63494bd",
            "test__md.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,4e1b42f49af7ed531d0d85fcae5effa8",
            "test__mode.nii.gz",
            "test__norm.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,e4f9a5a1ea4021f01879c2ba269b33dd",
            "test__rd.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,48d8030185bea2f3b11a0a648bbb6abd",
            "test__tensor.nii.gz",
            "test__nonphysical.nii.gz:md5:header,0968ff06e75bcbd0d7b2dcbcc5bf6ff7,data,4e865d417aae67c66e2a7c681aacfcae",
            "test__pulsation_std_dwi.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,62fe8dc99a6e3733c28b2c05281b6781",
            "test__pulsation_std_b0.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,e5bed827c14a49c4ff1aa97844ade64f",
            "test__residual.nii.gz:md5:header,c9e720b9a8acfe45a617e6486ebb8de0,data,a01c5897af1087af9a9153e20b491e91",
            [
                [
                    {
                        "id": "test"
                    },
                    "test__residual_iqr_residuals.npy:md5,58dfe3925469668723c16dc37f5c787c"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test__residual_mean_residuals.npy:md5,8a3c812ebbfc15b5242918d287659bf0"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test__residual_q1_residuals.npy:md5,2790db91a0e7727358dd2aaccb887983"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test__residual_q3_residuals.npy:md5,15c3d97be583a3e39bad3731aaea082d"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test__residual_residuals_stats.png:md5,bb819c98e3c98a8cd74cb1e79ae044d5"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test__residual_std_residuals.npy:md5,15fac15e7543f321a7d3a451e45563c5"
                ]
            ],
            [
                "versions.yml:md5,609a8b62c7dc862692683ec5860ba677"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-30T20:12:52.390001"
    }
}